frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { RefObject } from "react";
+import {
+  Box,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@chakra-ui/react";
+import { Resident } from "../../../types/ResidentTypes";
+import getFormattedDateAndTime from "../../../utils/DateUtils";
+
+type Props = {
+  residents: Resident[];
+  tableRef: RefObject<HTMLDivElement>;
+};
+
+const getFormattedDatesAndStatus = (resident: Resident) => {
+    const startDateObj = new Date(resident.dateJoined);
+    const startDate = getFormattedDateAndTime(startDateObj, true);
+
+    let endDate;
+    if (resident.dateLeft != null) {
+        const endDateObj = new Date(resident.dateLeft)
+        endDate = getFormattedDateAndTime(endDateObj, true)
+    }
+    const status = resident.dateJoined !== null && resident.dateLeft !== null ? "Past" : "Current";
+    return {
+        startDate,
+        endDate,
+        status
+    }
+}
+
+const ResidentDirectoryTable = ({
+  residents,
+  tableRef,
+}: Props): React.ReactElement => {
+  return (
+    <Box>
+      <TableContainer
+        marginTop="12px"
+        height="70vh"
+        overflowY="scroll"
+        ref={tableRef}
+      >
+        <Table
+          variant="baseTable"
+          minHeight="400px"
+          verticalAlign="middle"
+        >
+          <Thead>
+            <Tr>
+              <Th>Resident</Th>
+              <Th>Status</Th>
+              <Th>Building</Th>
+              <Th>Residency Start Date</Th>
+              <Th>Residency End Date</Th>
+            </Tr>
+          </Thead>
+
+          <Tbody>
+            {residents.map((resident) => {
+              const residentId = resident.initial + resident.roomNum;
+              const { startDate, endDate, status } = getFormattedDatesAndStatus(resident);
+                
+              return (
```

Comment:
noticing some weird behaviour in the table when there's only one item:
![image](https://github.com/uwblueprint/supportive-housing/assets/57375371/30a6bdf6-85d3-4c57-96fe-45d28e192d8b)

Seems to occur in log records as well, if you can find a fix for this (I think its just setting a max-height on each table row) awesome, if not I can make this a bug fix ticket

Reply:
with two residents:
![image](https://github.com/uwblueprint/supportive-housing/assets/57375371/0b8c1e4a-ad5a-4795-96c0-6c40097b1ab2)

Reply:
Resolved in this PR: https://github.com/uwblueprint/supportive-housing/pull/147